public/app/features/alerting/unified/components/rules/Filter/RulesFilter.v2.tsx

```diff
@@ -1,226 +1,769 @@
+              onBlur={() => {
+                const currentQuery = watch('query');
+                const parsedFilter = getSearchFilterFromQuery(currentQuery);
+                updateFilters(parsedFilter);
+              }}
+              value={watch('query')}
             />
-          </TabsBar>
-        }
-      >
-        <IconButton
-          name="filter"
-          aria-label={t('alerting.rules-filter.filter-options.aria-label-show-filters', 'Show filters')}
-        />
-      </PopupCard>
+          </div>
+          {/* the popup card is mounted inside of a portal, so we can't rely on the usual form handling mechanisms of button[type=submit] */}
+          <PopupCard
+            showOn="click"
+            placement="auto"
+            disableBlur={true}
+            isOpen={isPopupOpen}
+            onClose={() => setIsPopupOpen(false)}
+            onToggle={handleOnToggle}
+            content={
+              // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+              <div
+                ref={popupRef}
+                className={styles.content}
+                onClick={(e) => e.stopPropagation()}
+                onKeyDown={(e) => {
+                  if (e.key === 'Enter' || e.key === ' ') {
+                    e.stopPropagation();
+                  }
+                }}
+                role="dialog"
+                aria-label={t('alerting.rules-filter.filter-options.aria-label', 'Filter options')}
+                tabIndex={-1}
+              >
+                <FilterOptions
+                  onSubmit={handleAdvancedFilters}
+                  onClear={handleClearFilters}
+                  pluginsFilterEnabled={pluginsFilterEnabled}
+                />
+              </div>
+            }
+          >
+            <Button name="filter" icon="filter" variant="secondary" aria-label={filterButtonLabel}>
+              {filterButtonLabel}
+            </Button>
+          </PopupCard>
+          <RulesViewModeSelector viewMode={viewMode} onViewModeChange={onViewModeChange} />
+        </Stack>
+      </Stack>
+    </form>
+  );
+}
+
+interface FilterOptionsProps {
+  onSubmit: SubmitHandler<AdvancedFilters>;
+  onClear: () => void;
+  pluginsFilterEnabled: boolean;
+}
+
+const FilterOptions = ({ onSubmit, onClear, pluginsFilterEnabled }: FilterOptionsProps) => {
+  const styles = useStyles2(getStyles);
+  const theme = useStyles2((theme) => theme);
+  const { filterState } = useRulesFilter();
+  const isManualResetRef = useRef(false);
+
+  // Create portal container to render dropdowns above the popup modal
+  const portalContainer = usePortalContainer(theme.zIndex.portal + 100);
+
+  const defaultValues = searchQueryToDefaultValues(filterState);
+
+  // Fetch namespace and group data from all sources
+  const { currentData: grafanaPromRules = [], isLoading: isLoadingGrafanaPromRules } =
+    alertRuleApi.endpoints.prometheusRuleNamespaces.useQuery({
```

Comment:
It might be better to use `prometheusApi` instead of this. It has better types for GMA and DMA